templates/hammer_root.yml.epp

```diff
@@ -0,0 +1,8 @@
+<%- |
+  Optional[String]                             $epp_username,
+  Optional[Variant[String, Sensitive[String]]] $epp_password,
+| -%>
+:foreman:
+  # Credentials. You'll be asked for the interactively if you leave them blank here
+  :username: '<%= if $epp_username { $epp_username } else { 'undef'} %>'
+  :password: '<%= if $epp_password { $epp_password } else { 'undef'} %>'
```

Comment:
I'm not sure `undef` is accepted as a valid value.

Reply:
Of course this is not `undef`, but the String `'undef'`.  And of course with this String the Application will not work.  In the [originating Template](https://github.com/theforeman/puppet-foreman/blob/master/templates/hammer_root.yml.erb) there was no Handling of `undef` at all.   Do you have a better Solution?